services/hierarchy/src/main/kotlin/VulnerabilityService.kt

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.services
+
+import org.eclipse.apoapsis.ortserver.dao.dbQuery
+import org.eclipse.apoapsis.ortserver.dao.tables.AdvisorJobsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.AdvisorResultsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.AdvisorResultsVulnerabilitiesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.AdvisorRunsIdentifiersTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.AdvisorRunsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.VulnerabilitiesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.advisor.VulnerabilityDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.IdentifiersTable
+import org.eclipse.apoapsis.ortserver.dao.utils.listCustomQuery
+import org.eclipse.apoapsis.ortserver.model.VulnerabilityWithIdentifier
+import org.eclipse.apoapsis.ortserver.model.runs.Identifier
+import org.eclipse.apoapsis.ortserver.model.runs.advisor.Vulnerability
+import org.eclipse.apoapsis.ortserver.model.util.ListQueryParameters
+
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.ResultRow
+
+/**
+ * A service to interact with vulnerabilities.
+ */
+class VulnerabilityService(private val db: Database) {
+    suspend fun listForOrtRunId(
+        ortRunId: Long,
+        parameters: ListQueryParameters = ListQueryParameters.DEFAULT
+    ) = db.dbQuery {
+        VulnerabilityDao.listCustomQuery(parameters, ResultRow::toVulnerabilityWithIdentifier) {
+            (
+                    VulnerabilitiesTable innerJoin
+                            AdvisorResultsVulnerabilitiesTable innerJoin
+                            AdvisorResultsTable innerJoin
+                            AdvisorRunsIdentifiersTable innerJoin
+                            AdvisorRunsTable innerJoin
+                            AdvisorJobsTable innerJoin
+                            IdentifiersTable
+                    ).select(
+                    VulnerabilitiesTable.id,
+                    VulnerabilitiesTable.externalId,
+                    VulnerabilitiesTable.summary,
+                    VulnerabilitiesTable.description,
+                    IdentifiersTable.type,
+                    IdentifiersTable.namespace,
+                    IdentifiersTable.name,
+                    IdentifiersTable.version
+                ).where { AdvisorJobsTable.ortRunId eq ortRunId }
```

Comment:
Indentation looks weird here. Is it correct?

Reply:
This is what my IntelliJ is doing with autoformatting.
As detekt is not complaining, I would guess that it is correct?
And yes, it also looks weird and wrong to my eye.